src/utils/makeReducer/index.ts

```diff
@@ -0,0 +1,13 @@
+import { match, Cases } from "../match"
+import { Tagged } from "../../types/Tagged"
+import { Reducer } from "redux"
+
+export type MakeReducerCases<S, A extends Tagged> = (s: S) => Cases<A, S>
+
+export const makeReducer = <S, A extends Tagged>(
+  makeReducerCases: MakeReducerCases<S, A>,
+  initialState: S,
+): Reducer<S, A> => (state, action) => {
```

Comment:
Change the `||`s to:
```suggestion
): Reducer<S, A> => (state = initialState, action) => {
```